chrome/browser/net/chrome_network_delegate.cc

```diff
@@ -84,6 +85,38 @@ namespace {
 
 bool g_access_to_all_files_enabled = false;
 
+const std::regex RE_JS("\\.js(?:\\?.+)*$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_CSS("\\.css$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_IMAGE("\\.(?:gif|png|jpe?g|bmp|ico)$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_FONT("\\.(?:ttf|woff)$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+const std::regex RE_HTML("\\.html?$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+
+content::ResourceType detectResourceType(std::string url)
```

Comment:
since it does not look like `ChromeNetworkDelegate` class job i've used just function in anonymous namespace.

Reply:
Please make the argument `const std::string& url` to avoid copying the string.